typescript/web/src/pages/api/stripe/webhooks.ts

```diff
@@ -87,6 +91,12 @@ const webhookHandler = async (req: NextApiRequest, res: NextApiResponse) => {
           throw new Error("No workspace found for this subscription");
         }
 
+        if (!(status in WorkspaceStatus)) {
```

Comment:
By using `in`, you're comparing `status` with the keys of `WorkspaceStatus`, not its values. You have to compare with the values of `WorkspaceStatus`.

Reply:
No. I'm comparing `status` against the keys for a good reason, as I'm doing `WorkspaceStatus[status]` after.
Why would I compare if the `status` is in the values of my enum ?

Reply:
Yes, you've used the key below, that's why I've added [a suggestion](https://github.com/labelflow/labelflow/pull/913/files#r816625316) on the corresponding line.

Except for very specific cases, with enums we store the value and compare against it, not the key. The key is just an alias to the value it's not guaranteed that they are equal (_spoiler, they can be different if you change the configuration of graphql-codegen or Prisma, which is what I had to do at some point in #685 for example_). We already have some parts in the code with different keys and values (e.g. `Tools`).

You can see TypeScript enums as a way to reference string or int values with constants instead of plain values, It's actually even just transpiled as a JS const object.
i.e. instead of the plain `"Active"` word if `WorkspaceStatus` was not an enum but a union like `"Active" | "Trialing" | "Incomplete" | ...`, you can use `WorkspaceStatus.Active`.
But it's not more than that, the key is not expected to be exposed outside of the code, that's the purpose of the value.

You could eventually support such requirements but it'd need some very strong guards making sure that the enum values matches the keys (since assuming that they are equals without it would break the _single-responsibility principle_). But at the end, it's just more complex than relying on the value.